tests/Feature/ApiTest.php

```diff
@@ -0,0 +1,512 @@
+    function testCreateFailDateTooYoung()
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members', [
+            'name' => 'g_test' . $name ,
+            'information' => '1234567891',
+            'dob' => '2018-09-27',
+            'position' => 'intern',
+            'phone' => '0965841492',
+            'gender' => 2,
+            ]);
+        $response->assertStatus(422)
+        ->assertJson([
+            'message' => 'The given data was invalid.',
+            'errors' => ['dob' => ['The date canot early than today']]
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     */
+    function testUpdateMember($id)
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members/' . $id, [
+            'name' => 'test' . $name ,
+            'information' => '123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890',
+            'dob' => '1992-09-27',
+            'position' => 'intern',
+            'phone' => '0965841492',
+            'gender' => 1,
+            '_method' => 'PUT'
+            ]);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Member Updated Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     */
+    function testUpdateFailNameDatePhoneGenderPositionMember($id)
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members/' . $id, [
+            'name' => 'test_' . $name ,
+            'information' => '1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901',
+            'dob' => '1992-09-270',
+            'position' => 'intern1',
+            'phone' => '0965841492@',
+            'gender' => 3,
+            '_method' => 'PUT'
+            ]);
+        $response->assertStatus(422)
+        ->assertJson([
+            'message' => 'The given data was invalid.',
+            'errors' => ['name' => ['The name format is invalid.'],
+            'dob' => ['The dob is not a valid date.'],
+            'position' => ['The selected position is invalid.'],
+            'phone' => ['The phone format is invalid.'],
+            'gender' => ['The selected gender is invalid.']
+            ]
+            ]);
+    }
+
+    function testGetAllProjects()
+    {
+        $response = $this->json('GET', '/api/projects');
```

Comment:
please check that the list contains the correct data

Reply:
I have do it